### ISSUE TITLE
"Map" concept

### ISSUE THREAD

**Issue by a3qz** [a3qz/hag#113]
A minimap is a bit of a stretch for us here, but I think something like a button you could press to move around just the camera instead of the character would be nice (not impacting fov, but letting you see the layout of past rooms and hallways), now that we have breadcrumbs/exploration.  

A sense of spacial awareness is somewhat missing in gameplay, and i think this might help.

**Comment by JohnathonNow** [comment]
Panning the camera wouldn't be hard. What would need to happen besides separating the rendering from the player's coordinates though includes:  
1. Fixing the key to be what is visible, not whatever the current logic is that has always been a little wonky.  
2. Fix the ladder to only show once it has been found. Right now it is visible before being seen, and being able to pan the camera would make finding it too easy. Perhaps we could even add an item or potion that reveals the ladder before it is found, if we wanted.

**Comment by a3qz** [comment]
1. yeah the key stuff probably needs to be rewritten
2. is this still relevant given the recent changes to the camera?

**Comment by a3qz** [comment]
im thinking what would be nice here is implementing a "map mode" where when you have it enabled, all you can do is move with the movement keys.  we really just need to implement a separate concept of "thing that is focused on" and when you are in "control of the character" mode that thing's position is tied to the the user's location

**Comment by a3qz** [comment]
actually in general i like the idea of implementing a formal "input modes" system; this could allow us to do things like have more interactive menuing and such as well.

**Comment by JohnathonNow** [comment]
Yeah, we should have a context concept. It'll go nicely with restructuring the main function as well. We'd have to make some design decisions though, like whether moving in a map/menu counts as a turn, whether enemies and items should appear when not focused on the player, etc.
